# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

***Импортируем данные, изучим и подготовим их к анализу.***

*Для начала импортируем библиотеку Pandas и файл с данными для дальнейшей работы:*

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

*Выведем 20 строк датафрейма для изучения данных:*

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

***Вывод:***
<div style="border:solid green 1px; padding: 20px">
    
***В таблице содержится 12 столбцов:***  
**children** — количество детей в семье  
**days_employed** — общий трудовой стаж в днях  
**dob_years** — возраст клиента в годах  
**education** — уровень образования клиента  
**education_id** — идентификатор уровня образования  
**family_status** — семейное положение  
**family_status_id** — идентификатор семейного положения  
**gender** — пол клиента  
**income_type** — тип занятости  
**debt** — имел ли задолженность по возврату кредитов  
**total_income** — ежемесячный доход  
**purpose** — цель получения кредита  


 В столбце days_employed (общий трудовой стаж в днях) содержатся отрицательные числа, что не соотвествует значению столбца.  
 В столбце education (уровень образования клиента) для полноценного анализа нужно обработать неявные дубликаты - перевести шрифт к одному регистру.  
 В столбцах days_employed и total_income есть пропуски, их нужно заполнить медианным значением.  
 
</div>

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

*Выведем основную информацию о датафрейме:*

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


***Вывод:***
<div style="border:solid green 1px; padding: 20px">
В столбцах days_employed и total_income содержатся пропущенные значения типа NaN.
</div>

## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

*Выводим количество пропущенных значений для каждого столбца*

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Вывод:**
<div style="border:solid green 1px; padding: 20px">
В столбцах days_employed и total_income содержатся 2174 пропущенных значений. 
</div>

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

*Заполняем пропуски в столбце total_income медианным значением*

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

*Обработаем аномалии в столбце days_employed (общий трудовой стаж в днях) и переведем отрицательные значения в положительные*

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

*Выведем медианное значение для трудового стажа days_employed*

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

*Выведем уникальные значения столбца children(дети)*

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

*Удаляем строки с аномальными значениями в столбце children*

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

*Заполняем пропуски в столбце days_employed медианным значением для каждого типа занятости*

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

*Проверяем наличие пропусков*

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

**Вывод:**
<div style="border:solid green 1px; padding: 20px">
Возможные причины пропусков в данных связаны с тем, что, возможно, клиент не помнит свой трудовой стаж или он безработный.    
    
Все пропуски в датафрейме заполнены, можно приступать к дальнейшей обработке данных (поиск дубликатов, изменение вещественных чисел на цельночисленные).
</div>

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

*С помощью метода astype() изменим вещественные числа на целые в столбце total_income — ежемесячный доход*

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

*Обработаем неявные дубликаты. Приведем к нижнему регистру все значения в столбце education (уровень образования клиента)*

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

*Выводим количество строк-дубликатов и удаляем их.*

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

**Вывод:**
<div style="border:solid green 2px; padding: 20px">
Все данные из датафрейма обработаны. Приступаем к категоризации данных.
</div>

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

*В датафрейм добавим новый столбец total_income_category и распределим на категории кредитополучателей с разным уровнем дохода*

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

*Выведем уникальные цели для взятия кредита из столбца purpose (цель получения кредита)*

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

*Создадим новый столбец purpose_category и с помощью функции добавим новые категории, которые объединят цели получения кредита.*

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

In [22]:
data.head(10)

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   
5         0     926.185831         27    высшее             0   
6         0    2879.202052         43    высшее             0   
7         0     152.779569         50   среднее             1   
8         2    6929.865299         35    высшее             0   
9         0    2188.756445         41   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   
5  гражданский брак                 1      M   компаньон     0        255763   
6   женат / замужем                 0      F   компаньон     0        240525   
7   женат / замужем                 0      M   сотрудник     0        135823   
8  гражданский брак                 1      F   сотрудник     0         95856   
9   женат / замужем                 0      M   сотрудник     0        144425   

                      purpose total_income_category          purpose_category  
0               покупка жилья                     B  операции с недвижимостью  
1     приобретение автомобиля                     C    операции с автомобилем  
2               покупка жилья                     C  операции с недвижимостью  
3  дополнительное образование                     B     получение образования  
4             сыграть свадьбу                     C        проведение свадьбы  
5               покупка жилья                     B  операции с недвижимостью  
6           операции с жильем                     B  операции с недвижимостью  
7                 образование                     C     получение образования  
8       на проведение свадьбы                     C        проведение свадьбы  
9     покупка жилья для семьи                     C  операции с недвижимостью

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

**Сгруппируем значения столбцов children (дети) и столбец debt (задолженность по кредиту)**

In [23]:
children_grouped = data.groupby('children').agg({'debt': ['count', 'sum']})

*Выведем результат на экран и проверим работу группировки столбцов*

In [24]:
children_grouped

debt      
          count   sum
children             
0         14091  1063
1          4808   444
2          2052   194
3           330    27
4            41     4
5             9     0

*Посчитали долю просрочек по кредиту*

In [25]:
children_grouped['childen_result'] =  children_grouped['debt']['sum'] / children_grouped['debt']['count']

*Отсортируем выборку по столбцу childen_result в порядке убывания и проанализируем результат*

In [26]:
children_grouped.sort_values(by='childen_result', ascending=False)

debt       childen_result
          count   sum               
children                            
4            41     4       0.097561
2          2052   194       0.094542
1          4808   444       0.092346
3           330    27       0.081818
0         14091  1063       0.075438
5             9     0       0.000000

**Вывод:** 
<div style="border:solid SteelBlue 3px; padding: 20px">
Мы видим увеличение количества просроченных задолженностей по кредиту с увеличением количества детей. 
Бездетные реже уходят в просрочку, чем клиенты с детьми.
Клиенты с 4 детьми имеют самый высокий показатель по просроченным платежам за кредит.
Клиенты с 5 детьми вообще не имеют просроченной задолженности.
</div>

In [27]:
data.groupby('children').agg(число_клиентов=('debt','count'),
                             число_должников=('debt','sum'),
                             доля_должников=('debt','mean')) \
                        .style.format({'доля_должников' : '{:.2%}'})

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

**Сгруппируем данные по столбцу family_status (семейное положение) и debt (задолженность по кредиту)**


In [28]:
family_status_grouped = data.groupby('family_status').agg({'debt': ['count', 'sum']})

*Выведем результат на экран и проверим работу группировки столбцов*

In [29]:
family_status_grouped

debt     
                       count  sum
family_status                    
Не женат / не замужем   2796  273
в разводе               1189   84
вдовец / вдова           951   63
гражданский брак        4134  385
женат / замужем        12261  927

*Посчитали долю просрочек по кредиту*

In [30]:
family_status_grouped['family_status_result'] = family_status_grouped['debt']['sum'] / family_status_grouped['debt']['count']

*Отсортируем выборку по столбцу childen_result в порядке убывания и проанализируем результат*

In [31]:
family_status_grouped.sort_values(by='family_status_result', ascending=False)

debt      family_status_result
                       count  sum                     
family_status                                         
Не женат / не замужем   2796  273             0.097639
гражданский брак        4134  385             0.093130
женат / замужем        12261  927             0.075606
в разводе               1189   84             0.070648
вдовец / вдова           951   63             0.066246

In [32]:
data.groupby('family_status').agg(число_клиентов=('debt','count'),
                             число_должников=('debt','sum'),
                             доля_должников=('debt','mean')) \
                        .style.format({'доля_должников' : '{:.2%}'})

**Вывод:** 
<div style="border:solid SteelBlue 3px; padding: 20px">
Клиенты из категории не женат/не замужем и находящиеся в гражданском браке имеют больший процент просроченных платежей по кредиту. 
Те кто в разводе или овдовели чаще платят в срок, чем люди в браке.
</div>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

**Сгруппируем данные по столбцу total_income (ежемесячный доход) и debt (задолженность по кредиту)**

In [33]:
total_income_grouped = data.groupby('total_income').agg({'debt': ['count', 'sum']})

*Выведем результат на экран и проверим работу группировки столбцов*

In [34]:
total_income_grouped 

debt    
             count sum
total_income          
20667            1   1
21205            1   0
21367            1   0
21695            1   0
21895            1   0
...            ...  ..
1711309          1   0
1715018          1   0
1726276          1   0
2200852          1   1
2265604          1   0

[18504 rows x 2 columns]

*Посчитали долю просрочек по кредиту*

In [35]:
total_income_grouped['total_income_result'] = total_income_grouped['debt']['sum'] / total_income_grouped['debt']['count']

*Категоризируем уровень дохода создав  столбец total_income_id. Выполним целочисленное деление дохода на 50000, тем самым разбив людей на категории каждые 50000 дохода.*


In [36]:
def total_income_category(row):
    if row >= 500000:
        return 10
    else:
        row = row // 50000
        return row

data['total_income_id'] = data['total_income'].apply(total_income_category)
data['total_income_id'].value_counts()
total_income_grouped = pd.DataFrame()
total_income_grouped = data.groupby('total_income_id').agg({'debt': ['count', 'sum']})
total_income_grouped['total_income_result'] = total_income_grouped['debt']['sum'] / total_income_grouped['debt']['count']
total_income_grouped.sort_values(by='total_income_result', ascending = False)

debt      total_income_result
                count  sum                    
total_income_id                               
2                7110  619            0.087060
3                4738  403            0.085057
6                 617   51            0.082658
1                4073  331            0.081267
7                 329   24            0.072948
4                2242  162            0.072257
8                 195   13            0.066667
5                1323   88            0.066515
10                222   14            0.063063
0                 371   23            0.061995
9                 111    4            0.036036

**Вывод:** 
<div style="border:solid SteelBlue 3px; padding: 20px">
  На основании среза данных по уровню дохода и количеством просрочек по кредитам, можно сделать следующие выводы:  
    
- Люди с низким доходом имеют меньше просрочек по кредитам. Почти такой же показатель и у людей с очень высоким доходом.   

- Что касается людей с низким доходом. 
Кредит для них очень важен. Имея низкий доход, они взяли кредит на то, что действительно нужно и понимают, что важно не портить кредитную историю.  

- Что касается людей с очень высоким уровнем дохода. Они понимают ценность денег. 
И выплата кредита особо не влияет на их бюджет. Для них кредиты это постояннство, чаще всего связанное с бизнесом и портить кредитную историю им также не хочется.  
    
- Плохая история по просроченным платежам у категории с высоким доходом. 
Возможно есть не серьезное отношение к кредиту. И возможно в ближайшее время закроют кредит полностью.
  
- Самая "опасная" история у клиентов со средним достатком, возможно это связано с финансовой нагрузкой.
</div>

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

*Отстортируем данные по столбцу purpose (цель получения кредита)*

In [37]:
purpose_grouped=data.groupby('purpose_category').agg({'debt':['sum','count']})

*Выведем результат и проверим работу группировки столбцов*

In [38]:
purpose_grouped.head(10)

debt       
                          sum  count
purpose_category                    
операции с автомобилем    400   4279
операции с недвижимостью  780  10751
получение образования     369   3988
проведение свадьбы        183   2313

*Посчитали долю просрочек по кредиту*

In [39]:
purpose_grouped['purpose_result'] = purpose_grouped['debt']['sum'] / purpose_grouped['debt']['count']

In [40]:
purpose_grouped.sort_values(by='purpose_result', ascending=False)

debt        purpose_result
                          sum  count               
purpose_category                                   
операции с автомобилем    400   4279       0.093480
получение образования     369   3988       0.092528
проведение свадьбы        183   2313       0.079118
операции с недвижимостью  780  10751       0.072551

In [41]:
data.groupby('purpose_category').agg(число_клиентов=('debt','count'),
                             число_должников=('debt','sum'),
                             доля_должников=('debt','mean')) \
                        .style.format({'доля_должников' : '{:.2%}'})

**Вывод:** 
<div style="border:solid SteelBlue 3px; padding: 20px">
    
- Больше всего просрочек по кредитам на автомобиль и образование. 
Это объясняется тем, что на рынке автокредитования создаются привлекательные предложения и человек берет кредит не оценивая свои возможности. 
 
- Кредиты на проведение свадеб и на недвижимось имеют меньше просроченных платежей. 
Клиент, покупая недвижимость в ипотеку, хорошо просчитывает свои возможности, так как срок платежа очень большой и риск остаться без жилья после неуплаты кредита велик.
</div>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:*
Пропуски в данной таблице связаны с разными категориями граждан - у них не указан стаж и уровень дохода. 
Возможно, не указали по причине, что нет подтвержденного дохода и стажа (работа в черную и тд). Но этого мы в данном случае уточнить не можем.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Медиана позволит получить более точную информацию и снизить возможную статистическую погрешность.
Так как, если брать среднее значение, оно некорректно характеризует данные, когда некоторые значения сильно выделяются среди других.

### Шаг 4: общий вывод.

<div style="border:solid green 2px; padding: 20px">
    
Самыми надежными заемщиками будут - клиенты
без детей, в разводе или вдовцы/вдовы, имеющие низкий или очень высокий уровень дохода  и берущие деньги на недвижимость.
</div>    